ports/osqp/portfile.cmake

```
@@ -0,0 +1,48 @@
+if ((VCPKG_LIBRARY_LINKAGE STREQUAL dynamic) AND (CMAKE_HOST_SYSTEM_NAME STREQUAL Windows))
+    message(STATUS "${PORT} doesn't support building as dynamic library on Windows, overriding to static")
+    set(VCPKG_LIBRARY_LINKAGE static)
+endif()
```

Do not reinvent the wheel.
```suggestion
if(VCPKG_TARGET_IS_WINDOWS)
    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
endif()
```
and add `osqp:x64-windows-static=fail` to scripts/ci.baseline.txt`.

---

In conda-forge we have a patch to respect `BUILD_SHARED_LIBS` as usually done in C++ libraries packaged in vcpkg, I guess it could be useful here: https://github.com/conda-forge/libosqp-feedstock/blob/main/recipe/patches/0001-Add-OSQP_RESPECT_BUILD_SHARED_LIBS-option.patch .

---

If you want to use that patch, I am the author so I am totally fine re-licensing it for inclusion in vcpkg.

---

> respect `BUILD_SHARED_LIBS` as usually done in C++ libraries

That's a different topic. I didn't review the packages capabilities, only the port.

That patch should be either upstreamed, or trimmed to the minimal changes.

---

> That's a different topic. I didn't review the packages capabilities, only the port.

Sure, it was just a FYI as I guess @aboelens may be interested in it.

> That patch should be either upstreamed, or trimmed to the minimal changes.

The functionality is was kind of upstreamed (even if in the form of two independent libraries `OSQP_BUILD_SHARED_LIB` and `OSQP_BUILD_STATIC_LIB`), but in the 1.0 that has been in development since ~5 years and were all the changes are being done, but it still needs to be released.

---

@traversaro Thanks for the suggestion. I think for this port it makes sense to stay close to the original source code, but it will be a nice feature to have once 1.0 is released.

---

> osqp:x64-windows-static=fail

You are right. That lines is for port which only build dynamic libs.